### PR TITLE
Use bull for DTO conversion

### PR DESCRIPTION
**Pull request by nahguam**
# stream-registry PR

Use bull for DTO conversion

# PR Checklist Forms

- [ ] CHANGELOG.md updated
- [ ] Reviewer assigned
- [ ] PR assigned (presumably to submitter)
- [ ] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: graphql-api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/GraphQLDomain.java [hunk 1]
```diff
@@ -15,30 +15,18 @@
...
-import com.expediagroup.streamplatform.streamregistry.model.Domain;
-
 @Value
 @Builder
 public class GraphQLDomain {
   String name;
   String owner;
   String description;
-  Iterable<GraphQLKeyValue> tags;
+  List<GraphQLKeyValue> tags;
```
Comment (rshotels) [inline graphql-api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/GraphQLDomain.java]: Are we sure about this? Seems like tags should be a unique set.
Why the change from Iterable  to List, does BUL require this?

File: graphql-api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/GraphQLKeyValue.java [hunk 2]
```diff
@@ -29,23 +28,20 @@
...
-    return map
-        .entrySet()
+  public static List<GraphQLKeyValue> toMap(Map<String, String> map) {
+    return Optional
+        .ofNullable(map)
         .stream()
+        .flatMap(m -> m.entrySet().stream())
         .map(e -> new GraphQLKeyValue(e.getKey(), e.getValue()))
-        .collect(toList());
+        .collect(Collectors.toList());
```
Comment (rshotels) [inline graphql-api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/GraphQLKeyValue.java]: consider a static import

File: graphql-api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/GraphQLDomain.java (hunk 1 above)
Comment (nahguam) [reply graphql-api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/GraphQLDomain.java]: Yes, BUL requires the change in type. But yes you're right - it should be a Set with comparator on the key.

Comment (nahguam) [reply graphql-api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/GraphQLDomain.java]: We can look at that separately from this PR

File: graphql-api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/GraphQLKeyValue.java (hunk 2 above)
Comment (nahguam) [reply graphql-api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/GraphQLKeyValue.java]: ugh
